### PR TITLE
State diagram and clean of the app-fsm state machine

### PR DESCRIPTION
**Pull request by barroco**

### CODE REVIEW

File: js/app-fsm.js [hunk 1]
```diff
@@ -98,44 +93,59 @@ var appFsm = new machina.Fsm({
...
+        this.handle('getChannelList');
+
+      },
+      'getChannelList': function() {
+        storage.volatile.put('channel_list', [{name: 'BBC1', data:''}, {name:'BBC2', data:''}]);
+
         var self = this;
         setTimeout(function() {
-          storage.volatile.put('channel_list', [{name: 'BBC1', data:''}, {name:'BBC2', data:''}]);
+          if(!storage.volatile.get('channel_list') || storage.volatile.get('channel_list').length === 0) {
```
Comment (chrisn) [inline js/app-fsm.js]: Maybe simplify to avoid calling `storage.volatile.get('channel_list')` twice?

File: docs/app-fsm.puml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+AUTHORIZATION_CHECK: \t client_id:client_id
+AUTHORIZATION_CHECK: \t grant_type:'authorization_code'
+AUTHORIZATION_CHECK: \t code: 'device_code'
+AUTHORIZATION_CHECK: }
+
+
+'Transitions
+
+DEVICE_OFF -[#blue]-> DEVICE_ON: switch_on
+DEVICE_ON --> SCANNING
```
Comment (chrisn) [inline docs/app-fsm.puml]: Add a label for this transition?

Comment (barroco) [inline docs/app-fsm.puml]: Do you have a suggestion? it's a direct transition without trigger.

Comment (chrisn) [inline docs/app-fsm.puml]: I see, a label isn't necessary here.

**Pull request merged**